File: zain/valid_palindrome.py
def isPalindrome(self, s: str) -> bool:
    s = ''.join(c.lower() for c in s if c.isalnum())
    l, r = 0, len(s) - 1
    while l <= r:
        if s[l] != s[r]:
            return False
        l += 1
        r -= 1
    return True

File: ben/valid_palindrome.c
#include <stdbool.h>
#include <ctype.h>
#include <string.h>

bool isPalindrome(char * s){
    char* start = s;
    char* end = &s[strlen(s)] - 1;

    while (start < end) {
        if (!isalnum(*start)) {
            ++start;
            continue;
        }
        if (!isalnum(*end)) {
            --end;
            continue;
        }

        if (tolower(*start) != tolower(*end)) {
            return false;
        }
        ++start;
        --end;
    }
    return true;
}